comps/dataprep/redis/multimodal_langchain/prepare_videodoc_redis.py

```diff
@@ -0,0 +1,533 @@
+            file_prefix = os.path.splitext(video_file_name)[0]
+            if (file_prefix + ".vtt") not in captions_file_names:
+                raise HTTPException(
+                    status_code=400, detail=f"No captions file {file_prefix}.vtt found for {video_file_name}"
+                )
+
+        if len(video_files) == 0:
+            return HTTPException(
+                status_code=400,
+                detail="The uploaded files have unsupported formats. Please upload at least one video file (.mp4) with captions (.vtt)",
+            )
+
+        for video_file in video_files:
+            print(f"Processing video {video_file.filename}")
+
+            # Assign unique identifier to video
+            video_id = generate_video_id()
+
+            # Create video file name by appending identifier
+            video_name = os.path.splitext(video_file.filename)[0]
+            video_file_name = f"{video_name}_{video_id}.mp4"
+            video_dir_name = os.path.splitext(video_file_name)[0]
+
+            # Save video file in upload_directory
+            with open(os.path.join(upload_folder, video_file_name), "wb") as f:
+                shutil.copyfileobj(video_file.file, f)
+
+            # Save captions file in upload directory
+            vtt_file_name = os.path.splitext(video_file.filename)[0] + ".vtt"
+            vtt_idx = None
+            for idx, caption_file in enumerate(captions_files):
+                if caption_file.filename == vtt_file_name:
+                    vtt_idx = idx
+                    break
+            vtt_file = video_dir_name + ".vtt"
+            with open(os.path.join(upload_folder, vtt_file), "wb") as f:
+                shutil.copyfileobj(captions_files[vtt_idx].file, f)
+
+            # Store frames and caption annotations in a new directory
+            extract_frames_and_annotations_from_transcripts(
+                video_id,
+                os.path.join(upload_folder, video_file_name),
+                os.path.join(upload_folder, vtt_file),
+                os.path.join(upload_folder, video_dir_name),
+            )
+
+            # Delete temporary vtt file
+            os.remove(os.path.join(upload_folder, vtt_file))
+
+            # Ingest multimodal data into redis
+            ingest_multimodal(video_name, os.path.join(upload_folder, video_dir_name), embeddings)
+
+            # Delete temporary video directory containing frames and annotations
+            shutil.rmtree(os.path.join(upload_folder, video_dir_name))
+
+            print(f"Processed video {video_file.filename}")
+
+        return {"status": 200, "message": "Data preparation succeeded"}
+
+    raise HTTPException(
+        status_code=400, detail="Must provide at least one pair consisting of video (.mp4) and captions (.vtt)"
+    )
+
+
+@register_microservice(
+    name="opea_service@prepare_videodoc_redis", endpoint="/v1/dataprep/get_videos", host="0.0.0.0", port=6007
+)
+@traceable(run_type="tool")
```

Comment:
please remove langsmith traceable

Reply:
Thank for your feedback. We have removed langsmith here.